apps/wallet-dashboard/components/dialogs/staking/views/DetailsView.tsx

```diff
@@ -75,7 +76,9 @@ export function DetailsView({
     const [iotaEarnedFormatted, iotaEarnedSymbol] = useFormatCoin({ balance: iotaEarned });
     const [totalStakeFormatted, totalStakeSymbol] = useFormatCoin({ balance: totalStake });
 
-    const validatorName = validatorSummary?.name || '--';
+    const { data: inactiveValidatorSummary } = useGetInactiveValidator(validatorAddress);
+    const validatorName = inactiveValidatorSummary?.name || validatorSummary?.name;
```

Comment:
we should fallback to validator address so the code is in line with other similar components like StakedCard
`    const validatorName = inactiveValidatorSummary?.name || validatorSummary?.name || validatorAddress;
`